File: sdk-python/copilotkit/langchain.py
"""
copilotkit.langchain is deprecated. Use copilotkit.langgraph instead.
"""
import warnings
from copilotkit.langgraph import (
  langchain_messages_to_copilotkit,
  copilotkit_messages_to_langchain,
  copilotkit_customize_config,
  copilotkit_exit,
  copilotkit_emit_state,
  copilotkit_emit_message,
  copilotkit_emit_tool_call,
  copilotkit_interrupt,
)

warnings.warn(
    "copilotkit.langchain is deprecated. Use copilotkit.langgraph instead.",
    DeprecationWarning,
    stacklevel=2
)

__all__ = [
  "langchain_messages_to_copilotkit",
  "copilotkit_messages_to_langchain",
  "copilotkit_customize_config",
  "copilotkit_exit",
  "copilotkit_emit_state",
  "copilotkit_emit_message",
  "copilotkit_emit_tool_call",
  "copilotkit_interrupt",
]

File: packages/sdk-js/src/langchain.ts
console.warn(
  "Warning: '@copilotkit/sdk-js/langchain' is deprecated and will be removed in a future release. Please use '@copilotkit/sdk-js/langgraph' instead.",
);

export {
  CopilotKitPropertiesAnnotation,
  CopilotKitPropertiesSchema,
  CopilotKitStateAnnotation,
  CopilotKitStateSchema,
  type CopilotKitState,
  type CopilotKitSchemaState,
  type CopilotKitSchemaUpdate,
  type CopilotKitProperties,
  copilotkitCustomizeConfig as copilotKitCustomizeConfig,
  copilotkitExit as copilotKitExit,
  copilotkitEmitState as copilotKitEmitState,
  copilotkitEmitMessage as copilotKitEmitMessage,
  copilotkitEmitToolCall as copilotKitEmitToolCall,
  convertActionToDynamicStructuredTool,
  convertActionsToDynamicStructuredTools,
} from "./langgraph";
